src/javascript/app/pages/trade/purchase/contract-details.jsx

```diff
@@ -0,0 +1,219 @@
+                                                                    </span>
+                                                                    <span className='item-inner'>
+                                                                        {TimeTooltipWrapper(
+                                                                            <Text size='sm' bold={!!audit?.name}>{audit?.date}</Text>,
+                                                                            audit?.date
+                                                                        )}
+                                                                    </span>
+                                                                </div>
+                                                            </div>
+                                                        </React.Fragment>
+                                                    )}
+                                                </div>
+                                            </React.Fragment>
+                                        );
+                                    }
+
+                                    return false;
+                                })}
+                                <div className='table-box lg'>
+                                    <Explanation explanationOnly />
+                                </div>
+                            </div>
+                            <ContractTable data={data} />
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </>
+    );
+};
+
+const DetailsSection = ({ data }) => (
+    <>
+        <div className='header-box'>
+            <Button
+                variant='tertiary'
+                size='lg'
+                icon={<LabelPairedArrowLeftMdRegularIcon />}
+                color='black'
+                onClick={() => {
+                    dataManager.setPurchase({
+                        showContractDetailsPopup: false,
+                    });
+                }}
+            />
+            <div className='title-box'>
+                <Text size='md' bold>
+                    {localize('Contract Details')}
+                </Text>
+            </div>
+            <div className='spacer' />
+        </div>
+        <div className='popup-scroller'>
+            <div className='body-box'>
+                <div className='purchase-details-box'>
+                    <div className='details-title'>
+                        <Text centered size='md'>{data?.cd_description}</Text>
+                    </div>
+                    <div className='details-column'>
+                        <div className='chart-wrapper' id={data?.cd_chartId} />
+                        <ContractTable data={data} />
+                    </div>
+                </div>
+            </div>
+        </div>
+    </>
+);
```

Comment:
this file is overloaded with components, please separate them into separate files to increase readability🙏🏻